include/setup.h

```diff
@@ -111,26 +111,41 @@ class Value {
 
 class Property {
 public:
-	struct Changeable { enum Value {Always, WhenIdle,OnlyAtStart};};
+	struct Changeable {
+		enum Value { Always, WhenIdle, OnlyAtStart, Deprecated };
+	};
+
 	const std::string propname;
 
-	Property(std::string const& _propname, Changeable::Value when):propname(_propname),change(when) { }
+	Property(const std::string &name, Changeable::Value when)
+		: propname(name),
+		  value(),
```

Comment:
Is an empty 'name' string ok? could add a check in the constructor (I'm not sure if the origin code checked, but just thinking defensively).

Reply:
Yeah, we should prevent empty name - just in case. Perhaps I can code it in a way to test for this case at compile time - I think properties name are always hardcoded during definition.

Reply:
Ah, I went into the rabbit hole of trying to handle std::string in a `constexpr`… (this is impossible due to C++ design of std::string class). Decided to simply place an `assertm` in the constructor - if anyone will program empty property name, it will be obvious on the first test.